src/document/builders.js

```diff
@@ -251,7 +251,7 @@ function addAlignmentToDoc(doc, size, tabWidth) {
 }
 
 function label(label, contents) {
-  return { type: DOC_TYPE_LABEL, label, contents };
+  return label ? { type: DOC_TYPE_LABEL, label, contents } : contents;
```

Comment:
Why the label has to be `truthy`?

Reply:
For building code to be more compact. E.g.
```js
label(
  typeof doc.label === "object" && { commented: true, ...doc.label },
  [leading, doc, trailing]
)
```
No need to create an intermediate variable for `[leading, doc, trailing]`.

Reply:
If the `doc.label` is `null`, you can still separate in object, they result the same, `0`, `false`, `undefined`, '""' won't pass `typeof` check

Reply:
`doc.label` can't be `null` because the `label` builder doesn't accept falsy values. So this code adds a `type: "label"` doc wrapper only if `doc` is a label itself and its `.label` is an object. This way things like `.label.embed` and `.label.hug` are propagated. Otherwise it returns `[leading, doc, trailing]` without wrapping.

Reply:
I'm confusing myself.... you want to use the code above? Then the deleted comment seems more correct.

Reply:
I do use the code above, see the diff.

> you can still spread, they result the same

But I don't want to add the label doc wrapper at all in that case.

Reply:
> But I don't want to add the label doc wrapper at all in that case.

Make sense.

But when people call `label()` should expected a "label" doc to return.

Reply:
Compare these two alternatives. The first one looks better to me.
```js
// 1
function printTaggedTemplateLiteral(print) {
  const quasiDoc = print("quasi");
  return label(quasiDoc.label && { tagged: true, ...quasiDoc.label }, [
    print("tag"),
    print("typeParameters"),
    lineSuffixBoundary,
    quasiDoc,
  ]);
}

// 2
function printTaggedTemplateLiteral(print) {
  const quasiDoc = print("quasi");
  const resultDoc = [
    print("tag"),
    print("typeParameters"),
    lineSuffixBoundary,
    quasiDoc,
  ];
  return quasiDoc.label
    ? label({ tagged: true, ...quasiDoc.label }, resultDoc)
    : resultDoc;
}
```

Reply:
> But when people call label() should expected a "label" doc to return

There already are things like `join` among the doc builders that break this expectation. I think it's enough to document this behavior (which I did).

Reply:
The second one has more explicit intention to me.

Reply:
That's right, but the `concat` builder had more explicit intention than arrays too. I expect we're going to have more code like this (relying on propagated labels). It's good to be explicit in one-off scenarios, but not in something repeated.

Reply:
As a trade-off for explicitness, a `conditionalLabel` builder could be added, but I'm not convinced it's really needed.

Reply:
`concat` is a removed thing.

Reply:
Won't block this, just thought it's unnecessary limitation.

Reply:
> concat is a removed thing

Right. But it was explicit. It's just another instance of the typical "DRY vs explicit" tradeoff.

> Won't block this, just thought it's unnecessary limitation.

I saw a reusable pattern. I used it at least 3 times only in this PR and expect it to be used more. So I encapsulated it into the function.